ddl/cluster.go

```diff
@@ -96,6 +117,24 @@ func ValidateFlashbackTS(ctx context.Context, sctx sessionctx.Context, flashBack
 	return gcutil.ValidateSnapshotWithGCSafePoint(flashBackTS, gcSafePoint)
 }
 
+func setTiDBRestrictedReadOnly(sess sessionctx.Context, value bool) error {
+	var setValue string
+	if value {
+		setValue = variable.On
+	} else {
+		setValue = variable.Off
+	}
+	return sess.GetSessionVars().GlobalVarsAccessor.SetGlobalSysVar(variable.TiDBRestrictedReadOnly, setValue)
```

Comment:
Please notice that setting `tidb_restricted_read_only` to `on`, `tidb_super_read_only` is also applied, see:

https://docs.pingcap.com/tidb/stable/system-variables#tidb_restricted_read_only-new-in-v520

I would like to suggest using `tidb_super_read_only` instead.